tests/unit/resources/oauth.test.ts

```diff
@@ -0,0 +1,211 @@
+import { apply } from 'ruply';
+import createMollieClient, { OAuthGrantType, OAuthTokenType } from '../../..';
+import NetworkMocker, { getApiKeyClientProvider } from '../../NetworkMocker';
+
+const oauthEndpoint = { basePath: 'https://api.mollie.com:443', path: '/oauth2/tokens' };
+
+function decodeBasicAuth(authHeader: string): { clientId: string; clientSecret: string } {
+  const base64 = authHeader.replace('Basic ', '');
+  const decoded = Buffer.from(base64, 'base64').toString('utf-8');
+  const [clientId, clientSecret] = decoded.split(':');
+  return { clientId, clientSecret };
+}
+
+function parseFormUrlEncoded(body: string): Record<string, string> {
+  return Object.fromEntries(new URLSearchParams(body));
+}
+
+describe('oauth', () => {
+  describe('create', () => {
+    it('exchanges authorization code for tokens', () => {
+      return new NetworkMocker(getApiKeyClientProvider()).use(async ([mollieClient, networkMocker]) => {
+        let capturedBody: string | undefined;
+        let capturedHeaders: Record<string, string[]> | undefined;
+
+        networkMocker
+          .spy('POST', oauthEndpoint, (body, headers) =>
+            apply(
+              [
+                200,
+                {
+                  access_token: 'access_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+                  refresh_token: 'refresh_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
+                  expires_in: 3600,
+                  token_type: 'bearer',
+                  scope: 'payments.read payments.write',
+                },
+              ] as const,
+              () => {
+                capturedBody = body as string;
+                capturedHeaders = headers;
+              },
+            ),
+          )
+          .twice();
+
+        const token = await bluster(mollieClient.oauth.create.bind(mollieClient.oauth))({
+          clientId: 'app_xxxxxx',
+          clientSecret: 'secret_xxxxxx',
+          grant_type: OAuthGrantType.authorization_code,
+          code: 'auth_code_xxxxxx',
+          redirect_uri: 'https://example.com/callback',
+        });
+
+        // Verify response
+        expect(token.access_token).toBe('access_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx');
```

Comment:
nit: extract these expected strings into consts, so we can be sure they're identical and we don't have to count `x`s